### DISCUSSION TITLE
Handling the use-case of receipts vs invoices

### DISCUSSION THREAD

**Discussion by passatgt** [woocommerce/woocommerce#45006] [WooCommerce, Blocks, and Site Editing]
Hi! In Hungary, if you are selling virtual/digital products(like downloads, tickets for events), you have the option to send the customer "e-receipts". In this case, you only need a valid e-mail address to process a checkout, no need for a billing address. Currently with the normal checkout form, i can hide the extra fields by filtering `woocommerce_checkout_fields` and unsetting most of the billing fields, except the e-mail address. Obviously i'm not expecting a built-in solution for this, since its a niche problem, but something to keep in mind about extensibility, that this might be a use-case, because i don't think theres a way to do this right now.

**Comment by ralucaStan** [comment]
Adding here the conversation on the topic from [Developer Office Hours: November Edition](https://developer.woo.com/2023/11/02/developer-office-hours-november-edition/)

@passatgt : Also a similar "issue" is that i recently learned, that if a customer uses a local pickup shipping method(like a parcel locker), i can use the address of the parcel locker as the billing address, so in this case i don't need the customer's real address. So its a similar use-case to hide billing fields. 

@mikejolley  Is that local pickup use-case tied to payment methods somehow? Billing should match the card in some cases for example Stripe would error on mismatch. How is this handled?

@passatgt Depends on the provider i guess. In Hungary, cash on delivery is still very popular, so it doesn't matter in that case, and the popular card providers here do accept the payments with a valid name only. I just verified this on [muziker.de](http://muziker.de/) for example. As you can see i selected a parcel locker and it just shows the parcel locker address as billing address in readonly mode. And i can still pay with Stripe and even Apple Pay.

@alexflorisca Hmm this is interesting, we could maybe look at introducing the option of using the same address for billing as the store pickup location? Much like we have for using the shipping address as the billing address


@passatgt It gets complicated, because if you are a company, you do have to provide a valid billing address for invoicing. So i think just the ability to hide fields with code would be nice to have.

@mikejolley I think this may be one of the cases where the minor inconvenience of requiring address outweighs the potential conflicts with payment and other edge cases. But I agree billing should be customisable if you’re willing to accept that risk.

@passatgt Also another use-case for hiding billing fields: my extension shows a VAT number field in the first step below the e-mail address and it gets validated server side. The validation API returns the billing address and i can prefill the billing fields for accurate address. This do work, but if i could mark the fields readonly or hide them, that would be nice.

Conditionally hiding/showing payment and shipping methods is also frequently used with small code snippets, so an easy block/js equivalent for that would be nice. And also showing examples that one could use without setting up the dev environment would be nice, for example i recently learned i can just access
window.wc.blocksCheckout with js and do stuff with it
